src/smif/decision.py

```diff
@@ -24,38 +24,76 @@
+    writes the post-decision state through a DataHandle. This allows Models to access a given
+    decision state (identified uniquely by timestep and decision iteration id).
+
+    (Not yet implemented.) A DecisionManager may also pass a DataHandle down to a
+    DecisionModule, allowing the DecisionModule to access model results from previous timesteps
+    and decision iterations when making decisions.
 
     Arguments
     ---------
-    horizon: list
-    strategies: list
+    horizon: list strategies: list
     """
 
     def __init__(self, horizon, strategies):
         self._horizon = horizon
         self._strategies = strategies
+        self._decision_modules = []
 
-    def get_managers(self):
-        """Generate decision maker
-        """
-        decision_maker = None
+        self._set_up_decision_modules()
+
+    def _set_up_decision_modules(self):
+        interventions = []
 
         for strategy in self._strategies:
             if strategy['strategy'] == 'pre-specified-planning':
-                decision_maker = PreSpecified(
-                    self._horizon,
-                    strategy['interventions']
-                )
+                interventions.extend(strategy['interventions'])
             else:
                 msg = "Only pre-specified planning strategies are implemented"
                 raise NotImplementedError(msg)
 
-        if decision_maker is None:
-            decision_maker = PreSpecified(self._horizon, [])
+        self._decision_modules = [PreSpecified(self._horizon, interventions)]
+
+    def decision_loop(self):
+        """Generate bundles of simulation steps to run.
+
+        Each iteration returns a dict: {decision_iteration (int) => list of timesteps (int)}
 
-        yield (0, decision_maker)
+        With only pre-specified planning, there is a single step in the loop, with a single
+        decision iteration with timesteps covering the entire model horizon.
+
+        With a rule based approach, there might be many steps in the loop, each with a single
+        decision iteration and single timestep, moving on once some threshold is satisfied.
+
+        With a genetic algorithm, there might be a configurable number of steps in the loop,
+        each with multiple decision iterations (one for each member of the algorithm's
+        population) and timesteps covering the entire model horizon.
+
+        Implicitly, if the bundle returned in an iteration contains multiple decision
+        iterations, they can be performed in parallel. If each decision iteration contains
+        multiple timesteps, they can also be parallelised, so long as there are no temporal
+        dependencies.
+        """
+        assert len(self._decision_modules) == 1
+        assert isinstance(self._decision_modules[0], PreSpecified)
+        yield {0: self._horizon}
+
+    def get_state(self, timestep, iteration):
```

Comment:
Why is it not, get_intervention?

Reply:
Adding a line to the docstring - because this should return the system state, i.e. all interventions built in the given timestep for the given decision iteration.